src/rdb_protocol/terms/writes.cc

```diff
@@ -278,7 +278,7 @@ class replace_term_t : public op_term_t {
             rcheck(args->arg_is_deterministic(1) == deterministic_t::always,
                    base_exc_t::LOGIC,
                    "Could not prove argument deterministic.  "
-                   "Maybe you want to use the non_atomic flag?");
+                   "Maybe you want to use the non_atomic flag in the Update function");
```

Comment:
I think "the Update function" is somewhat ambiguous here. Especially since you can get this error from using not only `update`, but also `replace` and I think even `insert`. We can try to come up with a better solution for this one, or leave it out of the PR at first and focus on the `order_by` error message above.